continuous-testing/use-cases/markmap.md

```diff
@@ -1,56 +1,134 @@
+                * Using C/C++
+                    * CMock
+                    * cpp-stub
+                    * googletest
+    * For Code Analysis
+        * For static analysis
+            * For formatting, standards and structure
+                * Using Java
+                    * Checkstyle
+                    * Spotbugs
+                    * PMD
+                * Using Python
+                    * PyLint
+                    * Pyflakes
+                    * PMD
+                * Using C#
+                    * StyleCop
+                    * Gendarme
+                    * PMD
+                * Using C/C++
+                    * CppCheck
+                    * Clang Static Analyzer / tidy
+                    * PMD
+        * For dynamic analysis
+            * For test coverage
+                * Using Java
+                    * Jacoco
+                    * Cobertura
+                * Using Python
+                    * Coverage
+                    * PyTest-cov
+                * Using C#
+                    * OpenCover
+                    * NCover
+                * Using C/C++
+                    * Gcov
+                    * lcov
+            * For complexity analysis and runtime performance
+                * Using Java
+                    * JProfiler
+                    * JMeter
+                    * JPF
+                * Using Python
+                    * Scalene
+                    * sys.settrace()/inspect/pdb
+                * Using C#
+                    * Performance Profiler
+                    * Parasoft Ensure
+                    * ANTS Profiler
+                * Using C/C++
+                    * Valgrind
+                    * LLVM/Clang
+        * For security
+            * For vulnerability reporting and code checks
+                * Using most languages
+                    * Scrub
+                    * Veracode
     * For Web Applications
-        * Service Endpoints (APIs)
-            * Messaging-based
-                * RESTful Services
-                * SOAP
-                * Tools (JSON/XML)
-                    * Using Apache JMeter
-                    * Using Insomnia
-                    * Using Paw
-                    * Using Postman
-                    * Using REST-Assured (REST, RPC)
-                    * Using SOAPUI
-            * Procedure-based
-                * RPC
-                    * Using Shell/Powershell
-                    * Using PortQry
-        * Web Interfaces (browsers)
-            * Using Cypress
-            * Using Playwright
-            * Using Robot Framework
-            * Using Selenium + Add-on
-                * Cucumber* (Java + others) / Behave (Python) / Specflow (C#)
-                * Watir (Ruby)
-                * Home Grown
-    * For Backend (Load testing?)
-        * Using Chaos Monkey
-        * Using Robot Framework
-        * Using Cucumber Open
-        * For Mobile
+        * For service endpoints (APIs)
```

Comment:
This applies to all the tools / solutions listed - but have these actually been tested? How and why are we recommending them?

There's two criteria I'd recommend:
1. Down-select to the tools that you'd _personally_ recommend or vouch for. I think our use cases diagrams should not really be completely comprehensive for every possible tool / solution available, but rather the _best practices_ or _best tools_ we think we'd recommend to our community members.
2. Perform a trade-study on the remaining tools available, especially if they have similar proposed functionality, to help differentiate them

Reply:
Thanks. Changes made.